src/main/java/UserDetails.java

```diff
@@ -0,0 +1,88 @@
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+/**
+ * The UserDetails class provides methods to retrieve user details such as name, birthday, and gender.
+ */
+public class UserDetails {
+    private static final Scanner scanner = new Scanner(System.in);
+    private static final Pattern datePattern = Pattern.compile("^\\d{2}/\\d{2}/\\d{4}$");
+
+    /**
+     * Prompts the user to enter their full name and validates the input.
+     *
+     * @return the user's full name
+     */
+    public static String getName() {
```

Comment:
Naming of variables are good as it follows the naming conventions of using camelCase.